File: cpp/96.cpp
class Solution {
public:
  long nCm(int n, int m) {
    long ret = 1;
    for (int i = 0; i < m; i++) {
      ret *= (n - i);
      ret /= (i + 1);
    }
    return ret;
  }
  int numTrees(int n) { return nCm((2 * n), n) / (n + 1); }
};

File: go/96.go
package main

func nCm(n float64, m float64) int {
	var ret float64 = 1
	var i float64
	for i = 0; i < m; i++ {
		ret *= (n - i)
		ret /= (i + 1)
	}
	return int(ret)

}

func numTrees(n int) int {
	return nCm(float64(2*n), float64(n)) / (n + 1)
}
